driver/src/main/java/org/neo4j/driver/internal/async/NetworkConnection.java

```diff
@@ -61,12 +70,13 @@ public class NetworkConnection implements Connection {
     private final ExtendedChannelPool channelPool;
     private final CompletableFuture<Void> releaseFuture;
     private final Clock clock;
-
-    private final AtomicReference<Status> status = new AtomicReference<>(Status.OPEN);
     private final MetricsListener metricsListener;
     private final ListenerEvent<?> inUseEvent;
 
     private final Long connectionReadTimeout;
+
+    private Status status = Status.OPEN;
+    private UnmanagedTransaction transaction;
```

Comment:
`UnmanagedTransaction` depends on `Connection` and `Connection` is depending on `UnmanagedTransaction`, this makes theses classes highly coupled. 

The only part of the transaction needed in the connection is the method `executeWithLockedState`. Maybe you can extract an interface called `LockedStateExecutor` used it instead of the `UnmanagedTransaction`. So, the `bindTransaction` method will also change to `bindLockedStateExecutor` or something simillar. 

This also make the Connection/NetworkConnection more testable, since the code won't depend on a concrete implementation.

Reply:
Indeed. There is no need for full `UnmanagedTransaction` and it was not like that initially. During the investigation and on-going development the full transaction was eventually exposed. I have refactored that bit now. Thank!